src/main/java/no/nav/familie/integrasjoner/client/soap/PersonSoapClient.kt

```diff
@@ -56,28 +66,47 @@ class PersonSoapClient(private val port: PersonV3) : AbstractSoapClient("personV
-                                   "TPS.hentPersonhistorikk",
-                                   OppslagException.Level.LAV,
-                                   HttpStatus.FORBIDDEN,
-                                   e)
-        } catch (e: HentPersonhistorikkPersonIkkeFunnet) {
-            throw OppslagException("Prøver å hente historikk for person som ikke finnes i TPS",
-                                   "TPS.hentPersonhistorikk",
-                                   OppslagException.Level.LAV,
-                                   HttpStatus.NOT_FOUND,
-                                   e)
         } catch (e: Exception) {
-            throw OppslagException("Ukjent feil fra TPS",
-                                   "TPS.hentPersonhistorikk",
-                                   OppslagException.Level.KRITISK,
-                                   HttpStatus.INTERNAL_SERVER_ERROR,
-                                   e)
+            when {
+                e is HentPersonSikkerhetsbegrensning -> {
+                    throw OppslagException("Ikke tilgang til å hente personhistorikkinfo for person",
+                                           "TPS.hentPersonhistorikk",
+                                           OppslagException.Level.LAV,
+                                           HttpStatus.FORBIDDEN,
+                                           e)
+                }
+                e is HentPersonhistorikkPersonIkkeFunnet -> {
+                    throw OppslagException("Prøver å hente historikk for person som ikke finnes i TPS",
+                                           "TPS.hentPersonhistorikk",
+                                           OppslagException.Level.LAV,
+                                           HttpStatus.NOT_FOUND,
+                                           e)
+                }
+                sjekkOmExceptionErUnexpectedEOFinProlog(e) -> {
+                    throw OppslagException("Unexpected EOF in prolog",
+                                           "TPS.hentPersonhistorikk",
+                                           OppslagException.Level.MEDIUM,
+                                           HttpStatus.INTERNAL_SERVER_ERROR,
+                                           e)
+                }
+                else -> {
+                    throw OppslagException("Ukjent feil fra TPS",
+                                           "TPS.hentPersonhistorikk",
+                                           OppslagException.Level.KRITISK,
+                                           HttpStatus.INTERNAL_SERVER_ERROR,
+                                           e)
+                }
+            }
         }
     }
 
     override fun ping() {
         port.ping()
     }
+
+    private fun sjekkOmExceptionErUnexpectedEOFinProlog(e: Exception): Boolean {
+        return e is SoapFaultException &&
+               e.faultString != null &&
+               e.faultString.contains("Unexpected EOF in prolog")
```

Comment:
Er dette en kjent soap feil? Isåfall bør vel level kanskje nedjusteres til LAV?

Reply:
Like greit. Målet er å ikke forsøple loggene med error-meldinger.

Feilen er kjent og oppstår ved at nettverket klipper meldingen, som fører til en EOF. Ved en senere retry fikser det seg av seg selv.